test/functional/feature_config_args.py

```diff
@@ -145,12 +145,40 @@ def test_networkactive(self):
             self.start_node(0, extra_args=['-nonetworkactive=1'])
         self.stop_node(0)
 
+    def test_connect_with_seednode(self):
```

Comment:
<details><summary>test code suggestion</summary><p>

```python3
    def test_connect_with_seednode(self):
        self.log.info('Test -connect with -seednode')
        seednode_ignored = ['-seednode is ignored in -connect mode\n']

        # When -connect is supplied, expanding addrman via getaddr calls to ADDR_FETCH(-seednode)
        # nodes is irrelevant and -seednode is ignored.
        with self.nodes[0].assert_debug_log(expected_msgs=seednode_ignored):
            self.start_node(0, extra_args=['-connect=fakeaddress1', '-seednode=fakeaddress2'])

        # With -proxy, dns seeds are added to be used as ADDR_FETCH connections instead of resolving
        # to peers. ADDR_FETCH connections are not used in -connect mode.
        with self.nodes[0].assert_debug_log(expected_msgs=['-dnsseed is ignored in -connect mode when -proxy is specified\n']):
            self.restart_node(0, extra_args=['-connect=fakeaddress1', '-dnsseed=1', '-proxy=1.2.3.4'])

        for arg in ['-connect=0', '-noconnect']:
            with self.nodes[0].assert_debug_log(expected_msgs=[], unexpected_msgs=seednode_ignored):
                self.restart_node(0, extra_args=[arg, '-seednode=fakeaddress2'])
        self.stop_node(0)
```
</p></details>

- add logging
- use empty array for unneeded expected messages
- make more clear to the reader the differences between each test
- use the same "-seednode is ignore" message (not with/without '\n') for each test
- place the comments above the section rather than inside it (for readability)
- a bit shorter

(some of these are subjective)

Reply:
@jonatack Thank you for being so thorough! I learnt a lot here.

The only thing I didn't do was 

> use empty array for unneeded expected messages

because looking at [this code](https://github.com/bitcoin/bitcoin/blob/master/test/functional/test_framework/test_node.py#L388), I think that `assert_debug_log` would return too soon and not actually verify that the unexpected msgs are _not_ logged unless I also give it an expected message that would log _after_ the unexpected message. The underlying issue is how can the assert know that something did _not_ happen for sure? It either has to wait until the timeout, or until something that we are sure will happen after the unexpected thing _might_. 

Am I missing something?